### DISCUSSION TITLE
res.redirect() in /api, is not working

### DISCUSSION THREAD

**Discussion by umerjaved178** [vercel/next.js#34991] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

Chrome
Mac
Latest Nextjs version

### Which example does this report relate to?

https://nextjs.org/docs/api-routes/response-helpers#redirects-to-a-specified-path-or-url

### What browser are you using? (if relevant)

Chrome

### How are you deploying your application? (if relevant)

localhost

### Describe the Bug

I want to redirect user to my different page of the application, I am following this example https://nextjs.org/docs/api-routes/response-helpers#redirects-to-a-specified-path-or-url

I have made a button in frontend and upon clicking, I am fetching this but it is not redirecting my page

I am using fetching this (https://jsonplaceholder.typicode.com/posts), even if I remove the fetch completing, redirect is still not happening 

### Expected Behavior

It should redirect to res.redirect(`http://localhost:3000/features`) or to res.redirect(`/features`)

### To Reproduce

Create a nextjs app, 
1. create a button and onclicing call this api

**Comment by icyJoseph** [comment, marked as answer]
Hi,

This is a problem with how `fetch` works, which, I assume, is what you are using.

Simply put, the browser won't do `document` follow on a redirect originated from a fetch request. It will make the call though, so it is possible to be redirected to get more data, as far as I understand, but not to change the current document.

You can see this in Chrome Network tab, the type of your request is `fetch/Redirect`, and those won't do the trick. Why? Well if any script had the ability to just change the document like that, it'd be chaos. 

I think perhaps the docs, you point to, could use some more clarity, though I think it is ok, https://nextjs.org/docs/api-routes/response-helpers#redirects-to-a-specified-path-or-url.

> Taking a form as an example, you may want to redirect your client to a specified path or URL once they have submitted the form.

So there's the solution, do this (you could change the method to `GET` too):

```jsx
 <form method="POST" action="/api/redirect-endpoint">
        <button type="submit">Redirect</button>
 </form>
```

Further reading:

- https://stackoverflow.com/a/67479058
- https://blog.jim-nielsen.com/2021/fetch-and-3xx-redirect-status-codes/

**Reply by umerjaved178** [reply]
in short, you mean, res.redirect() only works if we make a POST request?

**Reply by icyJoseph** [reply]
No, the redirect, in this case, has to be started by a form. `fetch` is just like that.

**Reply by umerjaved178** [reply]
Thanks @icyJoseph 💥

**Reply by icyJoseph** [reply]
Don't forget to mark as answered so others can follow. It has happened that I ended up finding my own answer 😂.
